checklist/src/github.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2020 The AMP HTML Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Octokit} from '@octokit/rest';
+
+import {ILogger} from './types';
+
+/** Interface for working with the GitHub API. */
+export class GitHub {
+  /** */
+  constructor(
+    private client: Octokit,
+    private org: string,
+    private logger: ILogger = console
+  ) {}
+
+  /** Gets the directories added in a pull request inside a path. */
+  async getPullDirsAddedTo(
+    repo: string,
+    pull_number: number,
+    path: string
+  ): Promise<Array<string>> {
+    const {data: files} = await this.client.pulls.listFiles({
+      owner: this.org,
+      repo,
+      pull_number,
+    });
+    return files
+      .filter(({filename}) => new RegExp(`^${path}/.+/.+`).test(filename))
```

Comment:
Nit: This creates a new regexp object for every file in the iterable. Instead, I'd define

```javascript
const pathRegex = new RegExp(`^${path}/.+/.+`)
```
and use that. On another note, I wonder if some of this logic should live in the calling code. This function could be `getPullRequestFiles` and just return a list of filenames, and the ChecklistBot can own the logic for using that list of files.